src/Umbraco.Infrastructure/DependencyInjection/UmbracoBuilder.FileSystems.cs

```diff
@@ -49,7 +50,7 @@ internal static IUmbracoBuilder AddFileSystems(this IUmbracoBuilder builder)
                 ILogger<PhysicalFileSystem> logger = factory.GetRequiredService<ILogger<PhysicalFileSystem>>();
                 GlobalSettings globalSettings = factory.GetRequiredService<IOptions<GlobalSettings>>().Value;
 
-                var rootPath = hostingEnvironment.MapPathWebRoot(globalSettings.UmbracoMediaPhysicalRootPath);
+                var rootPath = Path.IsPathRooted(globalSettings.UmbracoMediaPhysicalRootPath) ? globalSettings.UmbracoMediaPhysicalRootPath : hostingEnvironment.MapPathWebRoot(globalSettings.UmbracoMediaPhysicalRootPath);
```

Comment:
Note that using `Path.IsPathRooted()` doesn't ensure an absolute path: it returns true for `C:Media`, but that is still relative to the current working directory (I guess that's the content root in an ASP.NET Core application). See: https://docs.microsoft.com/en-us/dotnet/api/system.io.path.ispathfullyqualified?view=net-5.0#remarks.

The same (wrong) assumption is made in the `PhysicalFileSystem`:
https://github.com/umbraco/Umbraco-CMS/blob/8eb3ef65f686619d09af478bd6fd563fc90580b2/src/Umbraco.Core/IO/PhysicalFileSystem.cs#L45-L54

Using `Path.IsPathFullyQualified()` seems more appropriate.

Reply:
It certainly seems that IsPathFullyQualified aligns better with the original intent in PhysicalFileSystem according to the comments, but I don't think it matters too much.


```csharp
var localRoot = @"C:\Users\Paul Johnson\Dev\umbraco\v9\src\Umbraco.Web.UI";
var physicalMediaPath = "C:media";
var combined = Path.Combine(localRoot, physicalMediaPath);
Path.GetFullPath(combined) == Path.GetFullPath(physicalMediaPath); // true
 ```